packages/storage/src/append_store.rs

```diff
@@ -239,6 +290,7 @@ where
     storage: &'a dyn Storage,
     start: u32,
     end: u32,
+    saved_indexes: HashMap<u32, Vec<Vec<u8>>>,
```

Comment:
The purpose is caching right?
1. Let's call it `cache`
2. Should we make it possible to turn it off to save gas when it's not useful?

Reply:
yes, can call it cache. I can't think of a user friendly way to turn it off, and I think this would save gas in every case where page size > 1 and there is more than one step of iteration